docs/man/cdt-cpp.1.md

```diff
@@ -0,0 +1,443 @@
+---
+title: CDT-CPP
+section: 1
+header: User Manual
+footer: cdt-cpp
+date: April 08, 2023
+---
+# NAME
+cdt-cpp - Eosio C++ to WebAssembly compiler
+
+# SYNOPSIS
+
+`cdt-cpp [options] <input file>`
+
+# DESCRIPTION
+
+**cdt-cpp**  Is a C++ compiler which converts C++ code of the smart contracts to WebAssemply binary code for 
+execution in a virtual machine of Antelope block chains  
+
+# OPTIONS
+
+**`-C`** 
+
+    Include comments in preprocessed output
+    
+**`--CC`** 
+
+    Include comments from within macros in preprocessed output
+    
+**`-D=<name definition>`** 
+
+    Define <macro> to <value> (or 1 if <value> omitted)
+
+**`-U=<name>`** 
+
+    Cancel any previous definition of name, either built in or provided with a -D option.              
+    
+**`-E`**
+
+    Only run the preprocessor    
+    
+**`-I=<dir>`**
+    
+    Add directory to include search path
+    
+**`-L=<dir>`**
+
+    Add directory to library search path    
+    
+**`--MF=<file>`**
+
+    Write depfile output    
+
+**`--MD`**
+
+    Write depfile containing user and system headers    
+    
+**`--MMD`**
+    
+    Like -MD except mention only user header files, not system header files. 
+    
+**`--MT=<target>`**
+
+    Specify name of main file output in depfile    
+    
+**`-O=<optimization_level>`**
+    
+    Optimization level s, 0-3     
```

Comment:
I concur, it is from `cdt-spp`'s help. I think the help is erroneous. I suspect this should read:
`-Os` or even `-O=<s>` instead of `-O=<string>`. 
I think it's kind of unclear as written. I suppose I expect to write `-O2` etc.

Reply:
> I concur, it is from `cdt-spp`'s help. I think the help is erroneous. I suspect this should read: `-Os` or even `-O=<s>` instead of `-O=<string>`. I think it's kind of unclear as written. I suppose I expect to write `-O2` etc.

@ScottBailey I can say even more: most of description in the --help text are useless or wrong, or the options don't work at all. It was the cause why I decided to stop writing of beautiful descriptions and make the man page with descriptions from --help. If I will test every option, research how it really works by its source code and then write its description it takes a lot of time. Instead of this I will fast create man pages with the descriptions from --help text, it allows to add them to building process, to installation packages, to documentation (that we have man pages) and anywhere they are needed as entities. And then I will update their content when it will be good for other tasks. In short we don't need to make the descriptions beautiful right now. I already have beautiful descriptions for many of these options I just rolled them back because too many options need to be tested and fixed.

Reply:
@jolly-fellow Did you write an issue to update the help? If not, let's write one that addresses our concerns so we can move on.